waku/v2/protocol/waku_archive/driver/postgres_driver.nim

```diff
@@ -0,0 +1,248 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/db_postgres,
+  std/sequtils,
+  std/parseutils,
+  std/strformat,
+  std/nre,
+  std/options,
+  stew/[results,byteutils],
+  std/strutils
+
+import
+  ../../../protocol/waku_message,
+  ../../../utils/time,
+  ../../../config,
+  ../common,
+  ../driver
+
+export postgres_driver
+
+type PostgresDriver* = ref object of ArchiveDriver
+  connection : DbConn
+  preparedInsert: SqlPrepared
+
+proc dropTableQuery(): string =
+  "DROP TABLE messages"
+
+proc createTableQuery(): string =
+  "CREATE TABLE IF NOT EXISTS messages (" &
+  " pubsubTopic VARCHAR NOT NULL," &
+  " contentTopic VARCHAR NOT NULL," &
+  " payload VARCHAR," &
+  " version INTEGER NOT NULL," &
+  " timestamp BIGINT NOT NULL," &
+  " id VARCHAR NOT NULL," &
+  " storedAt BIGINT NOT NULL," &
+  " CONSTRAINT messageIndex PRIMARY KEY (storedAt, id, pubsubTopic)" &
+  ");"
```

Comment:
As we aim to deploy this in a cloud environment and the DB will be shared, to avoid inconsistencies between nodes, the table and index creation (together with the DB migrations) should happen before the node upgrade, and it should be performed by the node operator (cc @jakubgs).

For testing purposes having it here may make sense, but it cannot be shipped as it is right now :)

Reply:
Indeed. It would at least require a flag to make this step optional.